react-components/src/query/useSearchMappedEquipmentFDM.tsx

```diff
@@ -60,15 +68,34 @@ export const useSearchMappedEquipmentFDM = (
       if (models.length === 0) {
         return [];
       }
-      const sources = createSourcesFromViews(viewsToSearch);
-      const chunkedSources = chunk(sources, 10);
-      if (chunkedSources.length === 0) {
-        chunkedSources.push([]);
+
+      const viewDefinitions = await queryClient.fetchQuery({
+        queryKey: [
+          'reveal',
+          'react-components',
+          'search-mapped-fdm',
+          'get-view-definitions-from-simple-source',
+          viewsToSearch
+        ],
+        queryFn: async () => {
+          const chunkedSources = chunk(viewsToSearch, 1000);
+          const views = [];
+          for (const chunk of chunkedSources) {
+            const res = await fdmSdk.getViewsByIds(createSourcesFromViews(chunk));
+            views.push(...res.items);
+          }
+          return views;
+        }
+      });
```

Comment:
Nice! Can we separate out this in a separate function, just to make this function a little less convoluted?

Reply:
Yep! Fixed in [69ce933](https://github.com/cognitedata/reveal/pull/5078/commits/69ce933edab88119bc93fe3cfdfa5d811ff3f83f)